src/ampJson.js

```diff
@@ -1,7 +1,14 @@
-export default function ampJson(data, component) {
+export default function ampJson(data, component, attributes={}) {
   if (!data || !component) return '';
+
+  const attributesString = attributes ?
+    Object.entries(attributes).map(([key, val]) => (
+      `${key}=${val}`
+    )).join(' ').trim() :
+    undefined;
+
   return Object.entries(data).map(([key, value]) => (
-    `<${component} id="${key}">
+    `<${component} id="${key}" ${attributesString}>
```

Comment:
@jeremyhuang3627 this may not be the most robust way, but for `component` `key` and `type` you might be able to use `encodeURIComponent` to sanitize their values. However, make sure to check that the output doesn't actually change. If it does you'll have to use a different method (or different values if possible).